parts/kubernetesmasteraddons-heapster-deployment.yaml

```diff
@@ -1,34 +1,103 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: heapster
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+  name: system:heapster-with-nanny
+rules:
+- apiGroups:
+  - extensions
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - get
+  - list
+  - watch
+  - update
+  - patch
+- apiGroups:
+  - ""
+  resources:
+  - events
+  - namespaces
+  - nodes
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: system:heapster-with-nanny
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:heapster-with-nanny
+subjects:
+- kind: ServiceAccount
+  name: heapster
+  namespace: kube-system
 ---
 kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  labels:
+    k8s-app: heapster
+    version: v1.4.0
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+  namespace: kube-system
+  name: heapster-v1.4.0
 spec:
   replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: heapster
+      version: v1.4.0
   template:
+    metadata:
+      labels:
+        k8s-app: heapster
+        version: v1.4.0
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ""
     spec:
+      tolerations:
+      - key: "CriticalAddonsOnly"
+        operator: "Exists"
+      serviceAccountName: heapster
       containers:
       - image: <kubernetesHeapsterSpec>
         command:
         - "/heapster"
         - "--source=kubernetes.summary_api:\"\""
         name: heapster
-        resources:
```

Comment:
I removed this as it's taken care of by the pod-resizer but thinking about it more and given some other issues we currently have with the resizer I agree. We should keep the resource limits. I'll put this back